fixcloudutils/redis/event_stream.py

```diff
@@ -185,6 +196,28 @@ async def _handle_stream_messages(self, messages: List[Any]) -> None:
                 # acknowledge all processed messages
                 await self.redis.xack(self.stream, self.group, *ids)
 
+    async def _handle_stream_messages_parallel(self, messages: List[Any], max_parallelesm: int) -> None:
+        """
+        Handle messages in parallel in unordered fasion. The number of parallel tasks is limited by max_parallelism.
+        """
+
+        async def handle_and_ack(msg: Any, message_id: StreamIdT):
+            await self._handle_single_message(msg)
+            await self.redis.xack(self.stream, self.group, message_id)
```

Comment:
Not sure how redis handles acknowledgment of messages arrived later, while previous messages are not done yet or even failing?

Reply:
If I understand the documentation correctly, redis should handle this well. I also wrote a small test, which tests the out of order acknowledgement and it works.